Index: src/AIModel/AIBase.cxx
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/AIModel/AIBase.cxx,v
retrieving revision 1.64.2.5
diff -U 3 -r1.64.2.5 AIBase.cxx
--- src/AIModel/AIBase.cxx	4 Apr 2007 09:51:43 -0000	1.64.2.5
+++ src/AIModel/AIBase.cxx	1 Jun 2007 03:48:08 -0000
@@ -324,6 +324,7 @@
 
 double FGAIBase::UpdateRadar(FGAIManager* manager) {
    double radar_range_ft2 = fgGetDouble("/instrumentation/radar/range");
+   double radar_range_nm =radar_range_ft2;
    bool force_on = fgGetBool("/instrumentation/radar/debug-mode", false);
    radar_range_ft2 *= SG_NM_TO_METER * SG_METER_TO_FEET * 1.1; // + 10%
    radar_range_ft2 *= radar_range_ft2;
@@ -409,8 +410,10 @@
      // horiz_offset += user_yaw; // FIXME: WHY WOULD WE WANT TO ADD IN SIDE-SLIP HERE?
 
      // calculate values for radar display
-     y_shift = range * cos( horiz_offset * SG_DEGREES_TO_RADIANS);
-     x_shift = range * sin( horiz_offset * SG_DEGREES_TO_RADIANS);
+     y_shift = (range * cos( horiz_offset * SG_DEGREES_TO_RADIANS)) / radar_range_nm;
+     x_shift = (range * sin( horiz_offset * SG_DEGREES_TO_RADIANS)) / radar_range_nm;
      rotation = hdg - user_heading;
      if (rotation < 0.0) rotation += 360.0;
      ht_diff = altitude_ft - user_altitude;
